### PR TITLE
Refactor code and correct gamma calculation

### LINKED ISSUE #89: BRNNs correction
In former implementation of BRNNs, the gamma parameter was not being correctly calculated. To do:

- [ ] Understand how gamma is actually computed.
- [ ] Implement the correct gamma computation.
- [ ] Test on regression, binary classification, and multi label classification tasks.
- [ ] Test Feature Importance analysis.

### PR DESCRIPTION
**Pull request by EdAguilarB**
## Description
Code refactor of the BRNNs models, with a big baseBRNN class with shared methods and with two children classes for regression and classification problems. The gamma parameters calculation has been corrected.

## Linked issues
- Closes #89

### CODE REVIEW

File: helix/machine_learning/models/BRNNs.py [hunk 1]
```diff
@@ -6,80 +6,46 @@
 from sklearn.utils.extmath import safe_sparse_dot
 
 
-class BRNNRegressor(MLPRegressor):
-    """A wrapper for the MLPRegressor class to create a BRNN regressor."""
+class BaseBRNNMixin:
```
Comment (dcl10) [inline helix/machine_learning/models/BRNNs.py:9]:
```suggestion
class BRNNMixin:
```

File: helix/machine_learning/models/BRNNs.py [hunk 2]
```diff
@@ -143,8 +94,8 @@ def log_loss(y_true, y_prob, sample_weight=None):
     return -np.average(xlogy(y_true, y_prob), weights=sample_weight, axis=0).sum()
 
 
-class BRNNClassifier(MLPClassifier):
-    """A wrapper for the MLPClassifier class to create a BRNN classifier."""
+class BRNNRegressor(BaseBRNNMixin, MLPRegressor):
```
Comment (dcl10) [inline helix/machine_learning/models/BRNNs.py:97]:
```suggestion
class BRNNRegressor(BRNNMixin, MLPRegressor):
```

File: helix/machine_learning/models/BRNNs.py [hunk 3]
```diff
@@ -168,66 +118,51 @@ def __init__(
...
+    def _compute_deltas(self, y, activations, deltas, n_samples):
+        errors = activations[-1] - y
+        rmse = self._compute_data_loss(y, activations)
+        deltas[-1] = errors / (n_samples * rmse)
 
-    def _activation_derivative_func(self):
-        from sklearn.neural_network._base import DERIVATIVES
 
-        return DERIVATIVES[self.activation]
+class BRNNClassifier(BaseBRNNMixin, MLPClassifier):
```
Comment (dcl10) [inline helix/machine_learning/models/BRNNs.py:134]:
```suggestion
class BRNNClassifier(BRNNMixin, MLPClassifier):
```

**Review by dcl10 — CHANGES_REQUESTED** [review]
That's all about all I can say for now. Perhaps @Grazziela nows more about the maths?
